RecoJets/JetProducers/plugins/BoostedTauSeedsProducer.cc

```diff
@@ -265,25 +289,25 @@ void BoostedTauSeedsProducer::produce(edm::Event& evt, const edm::EventSetup& es
       selectedSubjetPFCandidateAssociationForIsolation->insert(subjetRef2, pfCandidate);
     }
 
-    // find all PFCandidates that are constituents of the **other** subjet
-    std::vector<reco::PFCandidateRef> pfCandidatesInSubjet1 = getPFCandidates_exclJetConstituents(pfCandidates, subjetConstituents2, 1.e-4, true);
-    std::vector<reco::PFCandidateRef> pfCandidatesInSubjet2 = getPFCandidates_exclJetConstituents(pfCandidates, subjetConstituents1, 1.e-4, true);
-    if ( verbosity_ >= 1 ) {
-      std::cout << "#pfCandidatesInSubjet1 = " << pfCandidatesInSubjet1.size() << std::endl;
-      std::cout << "#pfCandidatesInSubjet2 = " << pfCandidatesInSubjet2.size() << std::endl;
-    }
-
-    BOOST_FOREACH( const reco::PFCandidateRef& pfCandidate, pfCandidatesInSubjet1 ) {
-      selectedSubjetPFCandidateAssociationForIsoDepositVetos->insert(subjetRef1, pfCandidate);
-    }
-    BOOST_FOREACH( const reco::PFCandidateRef& pfCandidate, pfCandidatesInSubjet2 ) {
-      selectedSubjetPFCandidateAssociationForIsoDepositVetos->insert(subjetRef2, pfCandidate);
-    }
+    //// find all PFCandidates that are constituents of the **other** subjet
+    //std::vector<reco::PFCandidateRef> pfCandidatesInSubjet1 = getPFCandidates_exclJetConstituents(pfCandidates, subjetConstituents2, 1.e-4, true);
+    //std::vector<reco::PFCandidateRef> pfCandidatesInSubjet2 = getPFCandidates_exclJetConstituents(pfCandidates, subjetConstituents1, 1.e-4, true);
+    //if ( verbosity_ >= 1 ) {
+    //  std::cout << "#pfCandidatesInSubjet1 = " << pfCandidatesInSubjet1.size() << std::endl;
+    //  std::cout << "#pfCandidatesInSubjet2 = " << pfCandidatesInSubjet2.size() << std::endl;
+    //}
+    //
+    //BOOST_FOREACH( const reco::PFCandidateRef& pfCandidate, pfCandidatesInSubjet1 ) {
+    //  selectedSubjetPFCandidateAssociationForIsoDepositVetos->insert(subjetRef1, pfCandidate);
+    //}
+    //BOOST_FOREACH( const reco::PFCandidateRef& pfCandidate, pfCandidatesInSubjet2 ) {
+    //  selectedSubjetPFCandidateAssociationForIsoDepositVetos->insert(subjetRef2, pfCandidate);
+    //}
```

Comment:
Hi,

this looks like the code for excluding the hadronic tau when computing the isolation of electrons and muons.
Please keep it.

Can you remind me how you compute the isolation of the muon in the mutau channel
in case dR(muon, tau) < 0.3, so that the hadronic tau decay products would spoil the isolation of the muon ?

Cheers,


Christian

On Feb 1, 2016, at 11:25 PM, cgalloni wrote:

> In RecoJets/JetProducers/plugins/BoostedTauSeedsProducer.cc:
> 
> > -      selectedSubjetPFCandidateAssociationForIsoDepositVetos->insert(subjetRef2, pfCandidate);
> > -    }
> > +    //// find all PFCandidates that are constituents of the **other** subjet
> > +    //std::vector<reco::PFCandidateRef> pfCandidatesInSubjet1 = getPFCandidates_exclJetConstituents(pfCandidates, subjetConstituents2, 1.e-4, true);
> > +    //std::vector<reco::PFCandidateRef> pfCandidatesInSubjet2 = getPFCandidates_exclJetConstituents(pfCandidates, subjetConstituents1, 1.e-4, true);
> > +    //if ( verbosity_ >= 1 ) {
> > +    //  std::cout << "#pfCandidatesInSubjet1 = " << pfCandidatesInSubjet1.size() << std::endl;
> > +    //  std::cout << "#pfCandidatesInSubjet2 = " << pfCandidatesInSubjet2.size() << std::endl;
> > +    //}
> > +    //
> > +    //BOOST_FOREACH( const reco::PFCandidateRef& pfCandidate, pfCandidatesInSubjet1 ) {
> > +    //  selectedSubjetPFCandidateAssociationForIsoDepositVetos->insert(subjetRef1, pfCandidate);
> > +    //}
> > +    //BOOST_FOREACH( const reco::PFCandidateRef& pfCandidate, pfCandidatesInSubjet2 ) {
> > +    //  selectedSubjetPFCandidateAssociationForIsoDepositVetos->insert(subjetRef2, pfCandidate);
> > +    //}
> 
> We don't used the code for the boosted taus, so I can remove it unless other analyses use it for private sample production. Do you know if somebody still use it? @veelken @isobelojalvo
> 
> �
> Reply to this email directly or view it on GitHub.
>